src/store/storage-controller.ts

```diff
@@ -42,16 +42,20 @@ export class StorageController {
     return messages;
   }
 
-  public static async delete(
+  /**
+   * Deletes a Record message.
+   */
+  public static async deleteRecordMessage(
     messageStore: MessageStore,
     dataStore: DataStore,
     tenant: string,
-    message: BaseMessage
+    message: GenericMessage
```

Comment:
Taking `GenericMessage` in a method called `deleteRecordMessage` feels a bit odd. I'd prefer if we either 1. did `message: RecordsWriteMessage` to make this method stricter, or 2. left the method name as `delete`.

Reply:
Agree it is a bit odd, I considered changing the type, but it will involve both `RecordsWriteMessage` and `RecordsDelete` message and in the future `RecordsCommit`, a union type here is super ugly and will also trigger cascading changes to the caller who only has `TimestampedMessage`. I also considered create a new type called `RecordsMessage`, but that'd not include `RecordsQueryMessage` and `RecordsReadMessage` so the name won't be accurate.

Because there is no non-record handler that calls this method (since they don't have data to delete), I renamed it to reflect that it is really meant for just record handlers. I was also considering to rename the class `RecordStorageController` as an alternative but that sounded unwieldy, so I passed.

I took your suggestion to rename back to delete, since it is the existing name so there is no regression. Let me know if you like any approaches above or have new ones!

Reply:
Agreed that all of those other approaches are not quite right. Keeping the name as `delete` is my favorite for now. Depending on how #406 shakes out, my fingers are crossed that we can change/remove `TimestampedMessage` and hopefully this `delete()` method will become easier to name and type.